--- a/cmake/modules/FindTBB.cmake
+++ b/cmake/modules/FindTBB.cmake
@@ -196,11 +196,11 @@ if(NOT TBB_FOUND)
 
   if(TBB_INCLUDE_DIRS)
     if (EXISTS "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h")
         file(READ "${TBB_INCLUDE_DIRS}/tbb/tbb_stddef.h" _tbb_version_file)
     else()
-        file(READ "${TBB_INCLUDE_DIRS}/tbb/version.h" _tbb_version_file)
+        file(READ "${TBB_INCLUDE_DIRS}/oneapi/tbb/version.h" _tbb_version_file)
     endif()
     string(REGEX REPLACE ".*#define TBB_VERSION_MAJOR ([0-9]+).*" "\\1"
         TBB_VERSION_MAJOR "${_tbb_version_file}")
     string(REGEX REPLACE ".*#define TBB_VERSION_MINOR ([0-9]+).*" "\\1"
         TBB_VERSION_MINOR "${_tbb_version_file}")
@@ -278,11 +278,11 @@ if(NOT TBB_FOUND)
   
   if(NOT MSVC AND NOT TBB_LIBRARIES)
     set(TBB_LIBRARIES ${TBB_LIBRARIES_RELEASE})
   endif()
 
-  set(TBB_DEFINITIONS "")
+  set(TBB_DEFINITIONS "TBB_VERSION_MAJOR=${TBB_VERSION_MAJOR}")
   if (MSVC AND TBB_STATIC)
     set(TBB_DEFINITIONS __TBB_NO_IMPLICIT_LINKAGE)
   endif ()
 
   unset (TBB_STATIC_SUFFIX)
